tests/test_workers/test_tasks/test_opm_operations.py

```diff
@@ -21,96 +21,80 @@ def mock_config():
         yield mc
 
 
-@mock.patch('iib.workers.tasks.opm_operations.socket')
-def test_get_free_port(mock_socket):
-    mock_socket.socket().bind.side_effect = [OSError(), OSError(), '']
-    mock_socket.poll.side_effect = [1, 1, None]
-    port = opm_operations._get_free_port(50051, 50054)
-    assert port == 50053
-
+def test_gen_port_for_grp():
+    conf = get_worker_config()
+    port_start = conf['iib_grpc_start_port']
+    port_end = port_start + conf['iib_grpc_max_port_tries']
+    port_range = set(range(port_start, port_end))
 
-@mock.patch('iib.workers.tasks.opm_operations.socket')
-def test_get_free_port_no_port(mock_socket):
-    mock_socket.socket().bind.side_effect = OSError()
-    with pytest.raises(IIBError, match='No free port has been found after 3 attempts.'):
-        opm_operations._get_free_port(50051, 50054)
+    with pytest.raises(IIBError, match='No free port has been found after 100 attempts.'):
+        for port in opm_operations._gen_port_for_grpc():
+            assert port in port_range
```

Comment:
IIUC this is testing the range from config which now is 100 but it can be changed, right?

Isn't it better to change the `match` to use the `iib_grpc_max_port_tries` instead of hardcoded 100?

Reply:
@JAVGan  where do you see hardcoded values? Maybe I am overlooking something or I am not sure what are you referring to.

Reply:
In the string `'No free port has been found after 100 attempts.`

This `100` can be changed, no?

Reply:
ah long day... yeah we can take it from config. I'll do it on Monday.

Reply:
please check